### PR TITLE
Feature/angular upgrade

### PR DESCRIPTION
**Pull request by safo6m**
Dependencies upgraded to support Angular 4.
Updated typescript and fixed linter errors.

### CODE REVIEW

**Comment by thiagopecanha** [comment]
@safo6m, Waiting for this PR be merged.

File: typings.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (HennerM) [inline typings.json]: do we really need typings for zone.js?
According to https://www.npmjs.com/package/@types/zone.js the package already contains type informations.

**Comment by HennerM** [comment]
Thanks for the pull request, looks good in general, just made one comment on zone.js Typings. I would appreciate if you can try to remove the extra typings.

**Comment by safo6m** [comment]
@HennerM You are right, unnecessary typings removed.

**Pull request merged**